File: src/app/conjugation/forms/past-short-negative.ts
import {Conjugation, Word, WordType} from "../conjugation";
import {NonPastShortNegative} from "./non-past-short-negative";

export class PastShortNegative implements Conjugation {
    getConjugation(word: Word): (Word | undefined) {
        switch (word.wordType) {

            // Verbs
            case WordType.GodanVerb:
            case WordType.IchidanVerb:
            case WordType.SuruVerb:
            case WordType.KuruVerb:
                return new NonPastShortNegative().getConjugation(word)?.replaceLastKana('かった')


            // Adjectives
            case WordType.IAdjective:
                return word.replaceLastKana('くなかった')

            case WordType.NaAdjective:
                return word.addSuffix('じゃなかった')


            default:
                return undefined;
        }
    }

    getTitle(): string {
        return "Past, short, negative"
    }

}

File: backend/app/conjugation/forms/past_short_negative.py
from ..core import Conjugation, Word, WordType
from .non_past_short_negative import NonPastShortNegative


class PastShortNegative(Conjugation):

    title = 'Past, short, negative'
    settings_title = 'Short, negative'

    def conjugate(self, word: Word) -> Word | None:
        match word.word_type:

            # Verbs
            case (
                WordType.GODAN_VERB
                | WordType.ICHIDAN_VERB
                | WordType.SURU_VERB
                | WordType.KURU_VERB
            ):
                negative = NonPastShortNegative().conjugate(word)
                if negative is None:
                    return None

                return negative.replace_last_kana('かった')

            # Adjectives
            case WordType.I_ADJECTIVE:
                return word.replace_last_kana('くなかった')

            case WordType.NA_ADJECTIVE:
                return word.add_suffix('じゃなかった')

            case _:
                return None
